src/ORM/Table.php

```diff
@@ -1542,14 +1552,27 @@ public function get(mixed $primaryKey, array $options = []): EntityInterface
         }
         $conditions = array_combine($key, $primaryKey);
 
-        $cacheConfig = $options['cache'] ?? false;
-        $cacheKey = $options['key'] ?? false;
-        $finder = $options['finder'] ?? 'all';
-        unset($options['key'], $options['cache'], $options['finder']);
+        if (is_array($finder)) {
+            // deprecationWarning(
```

Comment:
If the signature change is approved I'll uncomment this warning and update the tests.

Reply:
I think the signature looks great here. We don't appear to be giving up any backwards compatibility and are getting typed parameters as well.